packages/viewport/src/listener.native.js

```diff
@@ -25,21 +24,16 @@ const matchWidth = ( operator, breakpoint ) => {
 };
 
 const addDimensionsEventListener = ( breakpoints, operators ) => {
+	const operatorEntries = Object.entries( operators );
+	const breakpointEntries = Object.entries( breakpoints );
+
 	const setIsMatching = () => {
-		const matches = reduce(
-			breakpoints,
-			( result, width, name ) => {
-				Object.entries( operators ).forEach(
-					( [ operator, condition ] ) => {
-						const key = [ operator, name ].join( ' ' );
-						result[ key ] = matchWidth( condition, width );
-					}
-				);
-
-				return result;
-			},
-			{}
-		);
+		const matches = breakpointEntries.flatMap( ( [ name, width ] ) => {
```

Comment:
There's a missing `Object.fromEntries` here. The `setIsMatching` action expects an object.

Reply:
Good find. Would have been useful to have a test to catch this. 

Addressed in https://github.com/WordPress/gutenberg/pull/44572/commits/92e074eb50b77d0bbd9428fb04d51cc8510f4bfe